bibtex-actions.el

```diff
@@ -489,18 +489,19 @@ TEMPLATE."
   "Return citation keys at point as a list for `embark'."
   (when-let ((key (or (bibtex-actions-get-key-org-cite)
                       (bibtex-completion-key-at-point))))
-    (cons 'citation-key (if (listp key) key (list key)))))
+    ;; Currently, Embark only accepts a string. So we use a crm-like string to
+    ;; represent the list.
+    (cons 'citation-key (string-join (if (listp key) key (list key)) ","))))
```

Comment:
@minad - is the above what you mean by adapting this to use a crm string?

I'm getting this error, which means the list isn't passed to the function.

``` console
bibtex-completion-get-entry1: Wrong type argument: stringp, nil
```

Reply:
This looks correct. I cannot say more without a stack trace.

Reply:
> This looks correct. I cannot tell more without a stack trace.

Here's with point on a citation, with two references, and so two keys:

``` elisp
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  regexp-quote(nil)
  bibtex-completion-get-entry1(nil)
  bibtex-completion-get-entry(nil)
  bibtex-completion-open-url-or-doi((nil))
  bibtex-completion-open-pdf((nil) bibtex-completion-open-url-or-doi)
  bibtex-completion-open-any((nil))
  bibtex-actions-open((nil))
  funcall-interactively(bibtex-actions-open (nil))
  command-execute(bibtex-actions-open)
  #f(compiled-function () #<bytecode 0x11e8052d3efee9f9>)()
  embark--act(bibtex-actions-open "low2001,mcneill2011" t)
  embark-act(nil)
  funcall-interactively(embark-act nil)
  command-execute(embark-act)
```

Reply:
It seems the candidates are not injected correctly by Embark or lost somewhere on the way. But note that your target finder should separate the candidates with the `crm-separator`, in your case this seems to be `&`.

Reply:
> It seems the candidates are not injected correctly by Embark or lost somewhere on the way. 

Right. I will add this piece to the Embark issue then.

> But note that your target finder should separate the candidates with the crm-separator, in your case this seems to be &.

Yes, I thought about that, but wasn't sure whether that would be active outside of bibtex-read.

But in any case, I tried using & and got the same result.

Reply:
Did you try this with emacs -Q? Now that I think about it, there could be an issue with how `consult-completing-read-multiple` handles such `crm-separated` input.

Reply:
> Did you try this with emacs -Q? Now that I think about it, there could be an issue with how consult-completing-read-multiple handles such crm-separated input.

Yes, I used my test script, but commented out the `consult-completing-read-multiple` line.